src/agent/ICheqd.ts

```diff
@@ -3164,6 +4058,310 @@ export class Cheqd implements IAgentPlugin {
+
+			// fetch latest status list to get metadata
+			const statusList = await Cheqd.fetchBitstringStatusList({
+				credentialStatus: {
+					id: statusListCredential,
+				},
+			} as VerifiableCredential);
+
+			// For multi-purpose status lists, determine the appropriate statusPurpose
+			const statusPurpose = this.getStatusPurposeForMultiPurposeList(args.newStatus);
+
+			// construct credentials with proper status entries
+			args.credentials = args.updateOptions.statusListIndices.map((index, i) => ({
+				'@context': [],
+				issuer: args.updateOptions!.issuerDid,
+				credentialSubject: {},
+				credentialStatus: {
+					id: `${statusListCredential}#${index}`,
+					type: 'BitstringStatusListEntry',
+					statusPurpose: statusPurpose,
+					statusListIndex: `${index}`,
+					statusListCredential,
+					statusSize: statusList.metadata.statusSize || 1,
+					statusMessage: statusList.metadata.statusMessages || [],
+				},
+				issuanceDate: '',
+				proof: {},
+			}));
+		}
+		// if jwt credentials, decode them
+		const credentials = await Promise.all(
+			args.credentials.map(async (credential) =>
+				typeof credential === 'string' ? await Cheqd.decodeCredentialJWT(credential) : credential
+			)
+		);
+		// validate credentials - case: consistent issuer
+		if (
+			credentials
+				.map((credential) => {
+					return (credential.issuer as { id: string }).id
+						? (credential.issuer as { id: string }).id
+						: (credential.issuer as string);
+				})
+				.filter((value, _, self) => value && value !== self[0]).length > 0
+		) {
+			throw new Error('[did-provider-cheqd]: bulk update: Credentials must be issued by the same issuer');
+		}
+		// validate credentials - case: status list index uniqueness
+		if (
+			credentials
+				.map((credential) => credential.credentialStatus!.statusListIndex)
+				.filter((value, index, self) => self.indexOf(value) !== index).length > 0
+		) {
+			throw new Error('[did-provider-cheqd]: bulk update: Credentials must have unique status list index');
+		}
+		// validate credentials - case: status list credential consistency
+		const statusListCredentialUrl = credentials[0].credentialStatus?.statusListCredential;
+		if (!statusListCredentialUrl) {
+			throw new Error('[did-provider-cheqd]: bulk update: Invalid status list credential URL');
+		}
+		if (
+			!credentials.every(
+				(credential) => credential.credentialStatus?.statusListCredential === statusListCredentialUrl
+			)
+		) {
+			throw new Error('[did-provider-cheqd]: bulk update: Credentials must belong to the same status list');
+		}
+		// validate credentials - case: status list type
+		if (!credentials.every((credential) => credential.credentialStatus?.type === 'BitstringStatusListEntry')) {
+			throw new Error('[did-provider-cheqd]: bulk update: Invalid status list type');
+		}
+
+		// validate args in pairs - case: statusListFile and statusList
```

Comment:
These 3 conditions can be combined into a single if statement.

```
if (args.options?.publish && !args.fetchList && 
    !(args.options?.statusListFile || args.options?.statusList)) {
    throw new Error('publish requires statusListFile or statusList, if fetchList is disabled');
}
```

Reply:
These are independent pair checks (same as in other similar functions). The single combined check is in line 4213